pkg/operator/apiserver/audit/audit_policies_test.go

```diff
@@ -83,14 +84,59 @@ func TestGetAuditPolicy(t *testing.T) {
 			},
 			goldenFile: "oauth.yaml",
 		},
-		// TODO: add test with multiple customRules
+		{
+			name: "multipleCustomRules",
+			config: configv1.Audit{
+				Profile: "None",
+				CustomRules: []configv1.AuditCustomRule{
+					{
+						Group:   "system:authenticated:oauth",
+						Profile: "WriteRequestBodies",
+					},
+					{
+						Group:   "system:authenticated",
+						Profile: "AllRequestBodies",
+					},
+				},
+			},
+			goldenFile: "multipleCr.yaml",
+		},
+		{
+			name: "unknownProfile",
+			config: configv1.Audit{
+				Profile: "InvalidString",
+			},
+			errContains: "unknown audit profile \"InvalidString\"",
+		},
+		{
+			name: "unknownCustomRulesProfile",
+			config: configv1.Audit{
+				Profile: "None",
+				CustomRules: []configv1.AuditCustomRule{
+					{
+						Group:   "InvalidGroup",
+						Profile: "InvalidProfile",
+					},
+				},
+			},
+			errContains: "unknown audit profile \"InvalidProfile\" in customRules for group \"InvalidGroup\"",
+		},
 	}
 	for _, scenario := range scenarios {
 		t.Run(scenario.name, func(t *testing.T) {
 			// act
 			policy, err := GetAuditPolicy(scenario.config)
-			if err != nil {
-				t.Fatal(err)
+			if len(scenario.errContains) == 0 && err != nil {
+				t.Fatalf("Expected no error yet received error: %v", err)
+			}
+			if len(scenario.errContains) > 0 {
+				if err == nil {
+					t.Fatalf("Expected an error yet received no error: %v", err)
+
+				}
+				if scenario.errContains != err.Error() {
+					t.Errorf("Error is not expected error")
```

Comment:
`t.Errorf("Expected error message: %q, but got: %q", scenario.errContains, err.Error())` is more informative

Reply:
Resolved